transactions/src/main/java/org/jboss/as/txn/subsystem/TransactionSubsystem14Parser.java

```diff
@@ -203,9 +214,13 @@ protected void parseObjectStoreEnvironmentElementAndEnrichOperation(final XMLExt
             switch (attribute) {
                 case RELATIVE_TO:
                     TransactionSubsystemRootResourceDefinition.OBJECT_STORE_RELATIVE_TO.parseAndSetParameter(value, operation, reader);
+                    relativeToEncountered = true;
                     break;
                 case PATH:
                     TransactionSubsystemRootResourceDefinition.OBJECT_STORE_PATH.parseAndSetParameter(value, operation, reader);
+                    if (!value.equals(TransactionSubsystemRootResourceDefinition.OBJECT_STORE_PATH.getDefaultValue())) {
+                        relativeToEncountered = true;
```

Comment:
Should this variable be something like 'needsDefaultRelativeTo' with a default value of true and here and L217 sets it to false? I think a few years from now a future colleague in SET is going to see this and have hard time understanding why a value for PATH results in "relativeToEncountered = true". It took me a few minutes to figure it out and I'm looking at a diff where I understand the goal of the change very well.